Src/BootCamp.Chapter/Homework4.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Homework4
+    {
+        public static void Demo()
+        {
+            static void GatherInfo(string userNumber)
```

Comment:
This should not be nested in Demo method. Move it to Homework4 class. this makes code less readable, because you force to keep step function and implementation of a function that step is for to be in the same place. Which beats the purpose of adding a step function (other than reusability)